Prepare release 0.10.0beta2

notoraptor opened the pull request:
@nouiz 

- Date set to August 25th (tomorrow).
- Vision/State unchanged.
- NB: Check if cuDNN news are convenient !

nouiz commented on NEWS.txt:
```diff
@@ -3,182 +3,61 @@ Release Notes
- - Removed old GPU backend ``theano.sandbox.cuda``. New backend ``theano.gpuarray`` is now the official GPU backend
- - Support more debuggers for ``PdbBreakpoint``
+ - Support NumPy ``1.13``
+ - Faster optimization step with new destroy handler
```
Optional faster optimization step...

nouiz commented on NEWS.txt:
```diff
@@ -3,182 +3,61 @@ Release Notes
+   - Official support for ``v6.*`` and ``v7.*``, support for ``v5.*`` will be removed in next release
+   - Added spatial transformation operation based on cuDNN
+   - Updated and improved caching system for runtime-chosen cuDNN convolution algorithms
+   - Support cuDNN v7 tensor core operations for convolutions
```
with the {time,guess}_once and {time,guess}_on_shape_change algo.

notoraptor replied:
Indeed. And currently, it is even only for time_(once|on_shape_change) algorithms.

notoraptor commented:
Updated. I will wait tomorrow morning for rebasing.

abergeron commented on NEWS.txt:
```diff
@@ -3,182 +3,64 @@ Release Notes
 
-Theano 0.10.0beta1 (9th of August, 2017)
-========================================
+Theano 0.10.0beta2 (25th of August, 2017)
```
Maybe we want to update this?

notoraptor replied:
Indeed, I will also have to rebase this PR when we are ready to make the beta.

Which new date ?

notoraptor commented:
Rebased to current master branch and updated. I added a mention about conda recipe and I changed release date to `5th of September` (as I guess we won't do the release today, and we don't work monday [public holiday?]). Please see last commit https://github.com/Theano/Theano/pull/6353/commits/8cfb94dcb974f45cece4d080389f3161542b50b2 @nouiz @abergeron .

notoraptor commented:
I am rebasing this PR, but while testing theano with libgpuarray latest master branch, I still got `SKIP: pygpu disabled`. When checking pygpu info, I got this output:
```
(python2) C:\Users\notoraptor\mila\dev\git\theano>python
Python 2.7.13 |Continuum Analytics, Inc.| (default, May 11 2017, 13:17:26) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import pygpu
>>> import pygpu.version
>>> print(pygpu.__version__)
0.7.1+0.g8f21ba2a7.dirty
>>> print(pygpu.version.major, pygpu.version.minor, pygpu.version.patch)
(0, 6, 9)
>>>
```
So, `__version__` seems to be at least 0.7, but `version` module does not print expected values.

nouiz commented on NEWS.txt:
```diff
@@ -3,20 +3,21 @@ Release Notes
  - Bug fixes, crash fixes and warning improvements
 
-A total of 15 people contributed to this release since 0.10.0beta1, see list below.
+A total of 16 people contributed to this release since 0.10.0beta1, see list below.
```
I don't like having a list of people only for between beta1 and beta2. Can you just keep the full list since 0.9? Do you have a reason to have such a specific list here?

notoraptor replied:
No particular reason, I used to list contributors between only the two last released versions.

I will update the list with contributors since 0.9.

notoraptor commented:
Now I tried to compile and reinstall libgpuarray/pygpu from a fresh cloned repository, and I got a new error:
```
(python2) C:\Users\notoraptor\mila\dev\git\tmp>python
Python 2.7.13 |Continuum Analytics, Inc.| (default, May 11 2017, 13:17:26) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import pygpu
>>> import pygpu.version
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named version
>>> pygpu.version.major
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'version'
>>> from pygpu import version
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name version
>>> pygpu.__version__
u'0.7.1+0.g8f21ba2.dirty'
>>>
```

Module `pygpu.version` cannot be found, although variable `pygpu.__version__` is still available. Very strange.

notoraptor commented:
Rebased to integrate latests changes.

Now I will investigate errors related to pygpu version.

notoraptor commented:
Which stable version of theano should we mention in the doc ? It is currently `0.9` (associated with libgpuarray version `0.6.9`). Should it be changed to `0.10.0beta2` (with libgpuarray version `0.7.1`) ? If we tell to install pygpu via conda channel `mila-udem`, then it may be mandatory to update stable version mentioned in the doc (as channel `mila-udem` only contains pygpu `0.7.1`, which is not compatible with theano versions prior to `0.10.0beta2`).

See current doc here: http://deeplearning.net/software/theano_versions/dev/install_ubuntu.html#with-pip

@nouiz

nouiz commented:
The stable version is 0.9 until we release 0.10. The beta2 isn't a stable version.

We should mention in a note in the installation instruction how to install the beta2. Probably something like: `conda install -c mile-udem theano pygpu`.

notoraptor commented:
Channel `mila-udem` does not provide Windows packages for `theano`.

I suggest we mention `mila-udem`'s pygpu package (which is available on both Windows and Linux) only as alternative for non-stable Theano installations (ie. installations from Theano git repository, as these are the installations that require pygpu >= 0.7).

nouiz commented on doc/install_generic.inc:
```diff
@@ -91,6 +91,10 @@ libgpuarray
 `Step-by-step instructions <http://deeplearning.net/software/libgpuarray/installation.html#step-by-step-install>`__.
 
+Alternatively, we also provide a conda channel ``mila-udem`` which contains packages for ``pygpu`` and ``libgpuarray``
+that are compatible with the latest development version of Theano::
```
Tell only python 2.7 and 3.5

nouiz replied:
Currently, you need pygpu 0.7.1 that isn't in conda default channel. But you can install it with our own channel (that support only Python 2.7 and 3.5):

conda install -c mila-udem pygpu
pip install [--user] github... # For the latest dev version
pip install [--user] github... # For the beta2 version

nouiz commented on doc/install_generic.inc:
```diff
@@ -121,3 +125,8 @@ libgpuarray
 Install the latest, development version of libgpuarray following the
 `Step-by-step instructions <http://deeplearning.net/software/libgpuarray/installation.html#step-by-step-install>`__.
+
+Alternatively, we also provide a conda channel ``mila-udem`` which contains packages for ``pygpu`` and ``libgpuarray``
```
Remove the new stuff and write something like:

See note in the bleeding edge version about pygpu 0.7.1

notoraptor commented:
Release date updated.
